Lib/unittest/test/testmock/testwith.py

```diff
@@ -286,7 +286,7 @@ def test_mock_open_read_with_argument(self):
         # for mocks returned by mock_open
         some_data = 'foo\nbar\nbaz'
         mock = mock_open(read_data=some_data)
-        self.assertEqual(mock().read(10), some_data)
+        self.assertEqual(mock().read(10), some_data[:10])
```

Comment:
Should multiple calls to read return the following data as open does? Would be great getting somes tests for that. Example:

```
        some_data = 'foo\nbar\nbaz'
        mock = mock_open(read_data=some_data)
        self.assertEqual(mock().read(10), "foo\nbar\nba")
        self.assertEqual(mock().read(10), "z")
        self.assertEqual(mock().read(10), "")
```

Reply:
It already does, but you are calling `mock()` so it creates a fake file handle before each read call, 

You would need to do:

```
with mock() as f:
    f.read(10)
    f.read(10)
```

for example.

I'll push a commit with a test for this.